apps/api/src/app/events/dtos/trigger-event-request.dto.ts

```diff
@@ -45,7 +48,59 @@ export class TopicPayloadDto {
   type: TriggerRecipientsTypeEnum;
 }
 
-@ApiExtraModels(SubscriberPayloadDto, TenantPayloadDto, TopicPayloadDto)
+export class StepsOverrides {
+  @ApiProperty({
+    description: 'Passing the provider id and the provider specific configurations',
+    example: {
+      sendgrid: {
```

Comment:
Thinking out load here, would we want to use the integration identifier instead of provider id in case a User has 2 sendgrid integrations and he wants to override a specific one. 
On the same note if we would want to allow wildcard provider override this solution will be a good fit.

Reply:
That's a very good idea, I was initially concerned that it would be more complex and long, but it seems that the identifier defaults to the name. So it still makes sense.

One main downside i see, is that it's not how the framework works today. And also, it's not typescript autocomplete enabled.

Reply:
Giving it another thought, the user can't trigger to multiple providers in a single workflow, if he uses a providerOverride, than all the steps will use the same provider configuration 🤔